java/vortex-jni/src/main/java/dev/vortex/api/ScanOptions.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api;
+
+import org.immutables.value.Value;
+
+/**
+ * Create a new set of options for configuring the scan.
+ */
+@Value.Immutable
```

Comment:
one would hope that withers and records make immutables unnecessary but we are on java 11 here :(

Reply:
Whither a wither